source/slang/slang-check-constraint.cpp

```diff
@@ -949,6 +949,11 @@ namespace Slang
                     QualType(sndVectorType->getElementType(), snd.isLeftValue));
             }
         }
+
+        if (auto fstUniformParamGroupType = as<UniformParameterGroupType>(fst))
+            return TryUnifyTypes(constraints, QualType(fstUniformParamGroupType->getElementType(), fst.isLeftValue), snd);
+        if (auto sndUniformParamGroupType = as<UniformParameterGroupType>(snd))
+            return TryUnifyTypes(constraints, fst, QualType(sndUniformParamGroupType->getElementType(), snd.isLeftValue));
```

Comment:
Any reason we don't try to unify directly on `BuiltinGenericType`?

Reply:
An alternative is to use `PointerLikeType` here. `BuiltinGenericType` is too general as it is also  the base of `StucturedBufferTypeBase`, which shouldn't be automatically unfolded here.